mmrazor/models/algorithms/nas/dsnas.py

```diff
@@ -75,24 +71,18 @@ def __init__(self,
             estimator_cfg['type'] = 'mmrazor.ResourceEstimator'
         self.estimator = TASK_UTILS.build(estimator_cfg)
 
-        if fix_subnet:
-            # According to fix_subnet, delete the unchosen part of supernet
-            load_fix_subnet(self.architecture, fix_subnet)
-            self.is_supernet = False
-        else:
-            self.mutator = self._build_mutator(mutator)
-            # Mutator is an essential component of the NAS algorithm. It
-            # provides some APIs commonly used by NAS.
-            # Before using it, you must do some preparation according to
-            # the supernet.
-            self.mutator.prepare_from_supernet(self.architecture)
-            self.mutator.prepare_arch_params()
-
-            self.mutable_module_resources = self._get_module_resources()
-            self.search_space_name_list = list(
-                self.mutator._name2mutable.keys())
-
-            self.is_supernet = True
+        self.mutator = self._build_mutator(mutator)
+        # Mutator is an essential component of the NAS algorithm. It
+        # provides some APIs commonly used by NAS.
+        # Before using it, you must do some preparation according to
+        # the supernet.
+        self.mutator.prepare_from_supernet(self.architecture)
+        self.mutator.prepare_arch_params()
+
+        self.mutable_module_resources = self._get_module_resources()
+        self.search_space_name_list = list(self.mutator._name2mutable.keys())
+
+        self.is_supernet = True
```

Comment:
field "is_supernet" is not required.

Reply:
In dsnas, ‘is_supernet’ is used in lines 133/138/288/193 for the initialization of the finetune phase.